src/main.rs

```diff
@@ -83,6 +83,7 @@ fn main() {
 
         // OpenID Connect relying party endpoints
         get "/callback" => broker::handlers::oauth2::Callback::new(&app),
+        post "/callback" => broker::handlers::oauth2::Callback::new(&app),
```

Comment:
The OIDC spec says that `id_token` is supported with response modes `fragment` and `form_post`. Google only supports `fragment`, hence the `fragment_callback` trick.

The `GET` route here is repurposed for the `fragment_callback` trick, which is where users land with the `id_token` in the fragment. That page then copies fragment parameters to a form that calls the `POST` route here. (It's essentially a polyfill for `form_post` when the provider only supports `fragment`.)

I'm carefully avoiding calling the `GET` route here with the `id_token`, because it's dangerous according to the spec. I think that has to do with leaking the token in the `Referer` header.